src/plot_styles.py

```diff
@@ -1,46 +1,66 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 import os
-import shutil
 from gettext import gettext as _
 from pathlib import Path
 
 from cycler import cycler
 
-from gi.repository import Adw, GLib, Gtk
+from gi.repository import Adw, GLib, Gio, Gtk
 
 from graphs import file_io, graphs, misc, utilities
 
 
+def _styles_in_directory(directory):
+    enumerator = directory.enumerate_children("default::*", 0, None)
+    styles = {}
+    loop = True
+    while loop:
+        file_info = enumerator.next_file(None)
+        if file_info is None:
+            loop = False
+            continue
+        file = enumerator.get_child(file_info)
+        filename = file.query_info("standard::*", 0, None).get_display_name()
+        styles[Path(filename).stem] = file
+    enumerator.close(None)
+    return styles
+
+
 def get_system_styles(self):
-    path = os.path.join(self.pkgdatadir, "styles")
-    return {Path(file).stem: os.path.join(path, file) for file
-            in os.listdir(path)
-            if os.path.isfile(os.path.join(path, file))}
+    return _styles_in_directory(
+        Gio.File.new_for_uri("resource:///se/sjoerd/Graphs/styles"))
```

Comment:
It is generally recommended to work with the uri actually (so this is good), I've been trying to get that to work for added data as well. This is because just using the path itself may not always work in edge-cases.

The main edge case being when using the nautilus google drive plugin, but as the Flatpak sandbox does not grant access to it, I gave up on this. Think granting that access may ruin the "Safe" label on the Software Center as well, which I would like to keep.